### DISCUSSION TITLE
New table explorer UI interface discussion

### DISCUSSION THREAD

**Discussion by majin1102** [apache/amoro#3226] [Ideas]
Currently, Amoro's table exploration space uses a hover pop-up interaction way, which many users have reported to be unfriendly. Additionally, the separation of database and table into two different controls makes it inconvenient to search using a db.table string. It is suggested to implement a more user-friendly table exploration UI to address the following 4 issues:

1. Discontinue the use of hover-based interactions.
2. Integrate the display of databases and tables on a single page.
3. It is recommended to add more advanced search ways, such as table format.
4. Some table names may be  too long to be fully displayed. I suggest that the boundaries of the table explorer can be draggable.

I have found a mature product prototype for reference:
https://vimeo.com/617106036

![image](https://github.com/user-attachments/assets/29d154c3-0275-45ec-92c3-f126eb1ce21c)

This interface framework seems fitting Amoro well.

How do you guys feel cc @zhoujinsong @zhangmo8 @baiyangtx

**Comment by baiyangtx** [comment]
![image](https://github.com/user-attachments/assets/a02f29cc-8fb2-4228-bf2b-5041990f051b)

This place is not Format but Database.

If it is changed to Format, how should the Database be displayed?

**Reply by majin1102** [reply]
Database and table are two layer structure like:
![image](https://github.com/user-attachments/assets/bf7d5e6d-b310-4376-8cb6-86f48f1ac9a9)

**Reply by zhangmo8** [reply]
> Database and table are two layer structure like: ![image](https://private-user-images.githubusercontent.com/5731171/374787697-bf7d5e6d-b310-4376-8cb6-86f48f1ac9a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1Mzc0OTgsIm5iZiI6MTcyODUzNzE5OCwicGF0aCI6Ii81NzMxMTcxLzM3NDc4NzY5Ny1iZjdkNWU2ZC1iMzEwLTQzNzYtOGNiNi04NmY0OGYxYWM5YTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTBUMDUxMzE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2FhYTM1MWNlZjY4OTVhZWE3ZTBkZTM0NWNjZDYzYzg1N2FiNWY5MTk2M2ZjZjcwYWM4YjBkNmYxMTUwYjk3MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Kp291J3voZ8O0hvQVUv00yF42hpRmgplUycFlIE_9zs)

This tree hierarchy may cause other UI/UX issues

**Reply by minteliuwm** [reply]
> Database and table are two layer structure like: ![image](https://private-user-images.githubusercontent.com/5731171/374787697-bf7d5e6d-b310-4376-8cb6-86f48f1ac9a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg4NzE2MTgsIm5iZiI6MTcyODg3MTMxOCwicGF0aCI6Ii81NzMxMTcxLzM3NDc4NzY5Ny1iZjdkNWU2ZC1iMzEwLTQzNzYtOGNiNi04NmY0OGYxYWM5YTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTRUMDIwMTU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWRmMDZmYmNhZjU0MDI4MzQ0MjQ4ZWFjOGUxNmU5OGVkNTQzNGFkODA2Y2M1NGFlMzYyNzZlYWZkZjdhZjhhNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.3cY4ZMCP7Ug4iDhX0bcD2qF3m-ot2Yh9GK6Jeie-bMw)

Why the tree structure, can't we select Catalog in the header and then select Database? Also, do we need to have multiple tables open simultaneously?

**Reply by baiyangtx** [reply]
1. We do not need to have  multiple tables open simultaneously. 
2. We need search tables by `db.table` formated identifier.

**Comment by majin1102** [comment]
cc @minteliuwm

**Comment by zhangmo8** [comment]
I prefer this interaction

![image](https://github.com/user-attachments/assets/2fc85c55-6437-4de3-bcc8-7851bd0aedbf)
